lib/kryptonite/aes.ex

```diff
@@ -142,6 +142,26 @@ defmodule Kryptonite.AES do
     _, e -> {:error, e}
   end
 
+  @doc """
+  Encrypt a stream using AES in CTR mode.
+
+  ## Examples
+
+      iex> {key, iv} = {generate_key!(), Random.bytes!(16)}
+      iex> cypher = 'This is a secret' |> stream_encrypt(key, iv) |> Enum.to_list()
+      iex> is_list(cypher)
+      true
+  """
+  @spec stream_encrypt(Enumerable.t, key, iv) :: Enumerable.t
```

Comment:
Why not keeping the same name `encrypt_cbc` but pattern match on the input when it's an enumerable?
This way we could have just a bunch of functions all uniform `encrypt_cbc` and `encrypt_gcm` that adapt based on the input.